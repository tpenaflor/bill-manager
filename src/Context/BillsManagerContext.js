import React from 'react'

const BillsManagerContext = React.createContext()

export default BillsManagerContext