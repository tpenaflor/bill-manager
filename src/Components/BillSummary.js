import React from 'react'
import {useBillsManagerContext} from '../Context/BillsManagerContext'

import {participantBillSummary} from '../Selector/BillSelector'

const BillSummary = () => {
  const {participants, bills} = useBillsManagerContext()

  return (
    <div className="entry">
      <div> </div>
      {participants.map((part)=> (
        <div key={part.id}>
          <p>{`${part.name} total bill ${participantBillSummary(part.id, bills)}`}</p>
        </div>
      ))}
    </div>
  )
}

export default BillSummary