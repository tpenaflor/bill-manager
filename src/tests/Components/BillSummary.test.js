import React from 'react'
import {shallow} from 'enzyme'
import { shallowToJson  } from 'enzyme-to-json';

import BillSummary from '../../Components/BillSummary'
import * as BillsManagerContext from '../../Context/BillsManagerContext'
import {billFixture} from '../Fixtures/BillFixture'
import {participantFixture} from '../Fixtures/participantFixture'

test('show bill summary', ()=> {
    const contextValues = {
        bills : billFixture,
        participants : participantFixture
    }

    jest.spyOn(BillsManagerContext, 'useBillsManagerContext')
        .mockImplementation(() => contextValues)

    const wrapper = shallow(<BillSummary />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
})