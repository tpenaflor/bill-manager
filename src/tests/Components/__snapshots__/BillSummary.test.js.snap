// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show bill summary 1`] = `
<div
  className="entry"
>
  <div>
     
  </div>
  <div
    key="a4714460-a11c-4681-a21d-c7de2c57a02c"
  >
    <p>
      user1 total bill PHP 200.00
    </p>
  </div>
  <div
    key="ac33c426-53c4-4d38-ba0f-f8e569a9a507"
  >
    <p>
      user2 total bill PHP 2,100.00
    </p>
  </div>
  <div
    key="3712fdf0-5b62-41b6-9442-b35a41d5ec81"
  >
    <p>
      user3 total bill PHP 400.00
    </p>
  </div>
  <div
    key="6be42a94-61eb-47d4-8552-0369487f015c"
  >
    <p>
      user4 total bill PHP 0.00
    </p>
  </div>
</div>
`;
